src/main/java/com/laytonsmith/core/functions/Threading.java

```diff
@@ -575,15 +577,197 @@ public String getName() {
+		}
+
+		@Override
+		public Boolean runAsync() {
+			return null;
+		}
+
+		@Override
+		public Mixed exec(Target t, Environment environment, Mixed... args) throws ConfigRuntimeException {
+			String id = args[0].val();
+			Thread th = THREAD_ID_MAP.get(id);
+			if(th == null) {
+				return CBoolean.FALSE;
+			}
+			return CBoolean.get(th.isAlive());
+		}
+
+		@Override
+		public Version since() {
+			return MSVersion.V3_3_4;
+		}
+
+		@Override
+		public String getName() {
+			return "x_thread_is_alive";
+		}
+
+		@Override
+		public Integer[] numArgs() {
+			return new Integer[]{1};
+		}
+
+		@Override
+		public String docs() {
+			return "bool {mixed thread} Tests if this thread is alive. A thread is alive if it has been started and has not yet died.";
+		}
+	}
+
+	@api
+	@seealso({x_is_interrupted.class})
+	public static class x_interrupt extends AbstractFunction {
+
+		@Override
+		public Class<? extends CREThrowable>[] thrown() {
+			return new Class[0];
+		}
+
+		@Override
+		public boolean isRestricted() {
+			return true;
+		}
+
+		@Override
+		public Boolean runAsync() {
+			return null;
+		}
+
+		@Override
+		public Mixed exec(Target t, Environment environment, Mixed... args) throws ConfigRuntimeException {
+			String id = args[0].val();
+			Thread th = THREAD_ID_MAP.get(id);
+			if(th != null) {
+				th.interrupt();
+			}
+			return CVoid.VOID;
 		}
 
 		@Override
 		public Version since() {
 			return MSVersion.V3_3_4;
 		}
 
+		@Override
+		public String getName() {
+			return "x_interrupt";
+		}
+
+		@Override
+		public Integer[] numArgs() {
+			return new Integer[]{1};
+		}
+
+		@Override
+		public String docs() {
+			return "void {mixed thread} Interrupts this thread. In other words \"interrupted status\" set true."
+					+ " The \"interrupted status\" can be checked using the method: " + x_is_interrupted.class.getSimpleName() + "()"
+					+ " If this thread is blocked in an invocation of the " + Scheduling.sleep.class.getSimpleName() + "(), " + x_thread_join.class.getSimpleName() + "() methods"
+					+ " then its \"interrupt status\" will be cleared and it will receive an InterruptedException.";
+		}
 	}
 
+	@api
+	@seealso({x_interrupt.class})
+	public static class x_is_interrupted extends AbstractFunction {
+
+		private static java.lang.reflect.Method interrupt;
+		static {
+			try {
+				interrupt = Thread.class.getDeclaredMethod("isInterrupted", boolean.class);
+				interrupt.setAccessible(true);
+			} catch (Exception e) {
+				//never happened
+				throw new CancelCommandException(e.getMessage(), Target.UNKNOWN);
+			}
+		}
+
+		@Override
+		public Class<? extends CREThrowable>[] thrown() {
+			return new Class[0];
+		}
+
+		@Override
+		public boolean isRestricted() {
+			return true;
+		}
+
+		@Override
+		public Boolean runAsync() {
+			return null;
+		}
+
+		@Override
+		public Mixed exec(Target t, Environment environment, Mixed... args) throws ConfigRuntimeException {
+			Thread th;
+			boolean clearFlag = true;
```

Comment:
I believe that `x_is_interrupted(thread)` should not clear the interrupt flag by default. That should prevent users from accidentally clearing it, when not explicitly passing the argument.